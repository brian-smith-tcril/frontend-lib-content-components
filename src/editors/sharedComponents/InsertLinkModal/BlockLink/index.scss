.link-container {
  .title {
    overflow-wrap: break-word;
  }
}
