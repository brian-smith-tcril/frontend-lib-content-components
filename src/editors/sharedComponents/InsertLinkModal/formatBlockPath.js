/**
 * Formats a block path into title and subtitle.
 *
 * @param {string} path - The path to be formatted.
 * @returns {Object} - Formatted block path with title and subtitle.
 */
const formatBlockPath = (path) => {
  if (!path) {
    return {
      title: '',
      subTitle: '',
    };
  }
  const pathSlitted = path.split(' / ');
  let title = pathSlitted.pop();
  const subTitle = pathSlitted.join(' / ');

  if (!title.trim()) {
    // If the last part is empty or contains only whitespace
    title = pathSlitted.pop();
  }
  return {
    title,
    subTitle,
  };
};

export default formatBlockPath;
