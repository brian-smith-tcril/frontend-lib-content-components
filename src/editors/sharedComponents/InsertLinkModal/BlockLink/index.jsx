import PropTypes from 'prop-types';
import { Button } from '@openedx/paragon';
import { LinkOff } from '@openedx/paragon/icons';
import formatBlockPath from '../formatBlockPath';

import './index.scss';

const BlockLink = ({ path, onCloseLink }) => {
  const { title, subTitle } = formatBlockPath(path);
  return (
    <div className="link-container d-flex row p-4 rounded border border-gray-400 mx-4 mt-3">
      <div className="col-10">
        <p className="text-left">{subTitle}</p>
        <p className="h2 w-20 title">{title}</p>
      </div>
      <div className="col-2">
        <Button
          variant="tertiary"
          className="d-flex justify-content-center align-items-center"
          data-testid="close-link-button"
          size="lg"
          iconBefore={LinkOff}
          onClick={onCloseLink}
        >
         &nbsp;
        </Button>
      </div>
    </div>
  );
};

BlockLink.propTypes = {
  path: PropTypes.string.isRequired,
  onCloseLink: PropTypes.func.isRequired,
};

export default BlockLink;
