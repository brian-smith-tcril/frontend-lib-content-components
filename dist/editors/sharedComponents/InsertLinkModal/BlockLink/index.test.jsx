import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import formatBlockPath from '../formatBlockPath';
import BlockLink from './index';

describe('BlockLink Component', () => {
  const defaultProps = {
    path: 'Some Path',
    onCloseLink: jest.fn(),
  };

  const renderComponent = (overrideProps = {}) => render(
    <BlockLink {...defaultProps} {...overrideProps} />,
  );

  test('renders with default props', () => {
    renderComponent();
    expect(screen.getByText('Some Path')).toBeInTheDocument();
  });

  test('snapshot', () => {
    const { container } = renderComponent();
    expect(container).toMatchSnapshot();
  });

  test('renders correctly with custom path', () => {
    const customProps = {
      ...defaultProps,
      path: 'Custom Path',
    };
    renderComponent(customProps);
    expect(screen.getByText('Custom Path')).toBeInTheDocument();
  });

  test('calls onCloseLink when the button is clicked', () => {
    renderComponent();
    fireEvent.click(screen.getByTestId('close-link-button'));
    expect(defaultProps.onCloseLink).toHaveBeenCalledTimes(1);
  });

  test('renders with valid title and subtitle', () => {
    const customProps = {
      path: 'Root Section / Child 1',
      onCloseLink: jest.fn(),
    };

    renderComponent(customProps);
    const { title, subTitle } = formatBlockPath(customProps.path);

    expect(screen.getByText(title)).toBeInTheDocument();
    expect(screen.getByText(subTitle)).toBeInTheDocument();
  });
});
