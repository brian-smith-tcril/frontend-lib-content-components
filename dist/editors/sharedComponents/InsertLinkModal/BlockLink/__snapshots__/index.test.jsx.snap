// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockLink Component snapshot 1`] = `
<div>
  <div
    class="link-container d-flex row p-4 rounded border border-gray-400 mx-4 mt-3"
  >
    <div
      class="col-10"
    >
      <p
        class="text-left"
      />
      <p
        class="h2 w-20 title"
      >
        Some Path
      </p>
    </div>
    <div
      class="col-2"
    >
      <button
        class="d-flex justify-content-center align-items-center"
        data-testid="close-link-button"
        variant="tertiary"
      >
        Â 
      </button>
    </div>
  </div>
</div>
`;
